torch/_export/exported_program.py

```diff
@@ -334,29 +334,29 @@ def _process_constraints(
                     equality_constraints.append((node_dim, other_node_dim))
 
     # Create dict mapping symbol to a singular range (lower, upper)
-    range_constraints: Dict[sympy.Symbol, RangeConstraint] = {}
+    range_constraints: Dict[sympy.Symbol, ValueRanges] = {}
 
     # Add inline constraints to range_constraints
     for symbol, value_range in inline_constraints.items():
-        range_constraints[symbol] = RangeConstraint(value_range.lower, value_range.upper)
+        range_constraints[symbol] = ValueRanges(value_range.lower, value_range.upper)
 
     # Add input range constraints to range_constraintss
     for input_dim, multi_range_constraint in multi_range_constraints.items():  # type: ignore[assignment]
         # Simplify the range constraints into a single range constraint
         # Ex. ranges [2, 10] and [3, 11] would get merged to [3, 10]
-        min_vals = [rc.min_val for rc in multi_range_constraint]
-        max_vals = [rc.max_val for rc in multi_range_constraint]
-        min_val = max(min_vals)
-        max_val = min(max_vals)
-        assert min_val <= max_val
+        min_vals = [rc.lower for rc in multi_range_constraint]
+        max_vals = [rc.upper for rc in multi_range_constraint]
+        min_val = max(min_vals)  # type: ignore[type-var]
+        max_val = min(max_vals)  # type: ignore[type-var]
+        assert min_val <= max_val  # type: ignore[operator]
 
         # Add input node range constraints
         val = placeholder_nodes[input_dim.input_name].meta["val"]
         assert isinstance(val, FakeTensor)
         symint = val.shape[input_dim.dim]
         assert isinstance(symint, SymInt), f"Expected SymInt but got {symint}: {type(symint)}"
         symbol = symint.node._expr
-        range_constraints[symbol] = RangeConstraint(min_val, max_val)
+        range_constraints[symbol] = ValueRanges(min_val, max_val)
```

Comment:
See if you can simply use `&` on `ValueRanges`, which would be a much better way of doing this and also take advantage of this change.